mpegts/mpegts/mpegts_demuxer.cpp

```diff
@@ -111,7 +112,13 @@ int MpegTsDemuxer::decode(SimpleBuffer *in, TsFrame *&out)
                         break;
                     }
                 }
-                _ts_frames[ts_header.pid]->_data->append(in->data() + in->pos(), 188 - in->pos() - pos);
+                
+                if(_ts_frames[ts_header.pid]->expected_pes_packet_length != 0 && _ts_frames[ts_header.pid]->_data->size() + 188 - in->pos() - pos > _ts_frames[ts_header.pid]->expected_pes_packet_length) {
+                    _ts_frames[ts_header.pid]->_data->append(in->data() + in->pos(), _ts_frames[ts_header.pid]->expectedPESLength - _ts_frames[ts_header.pid]->_data->size());

```

Comment:
typo? expected_pes_packet_length

Reply:
Yes, copy and past error when moving my work to a fork... doh!